### DISCUSSION TITLE
useLayoutEffect

### DISCUSSION THREAD

**Discussion by honaseung** [honaseung/react-deep-dive#29] [Chapter3]
When I first met with this, I was so scared of this. `useLayoutEffect`? What it looks like a complicated concept of somethig. I thought. Now I think it's suck. As I know it's deprecated from formal document. So let's just look over it.

This is a similar with `useEffect` as it looks like. The only different point is 'when'. When does it happen is a quite different point of between `useEffect` and `useLayoutEffect`.

1. React DOM update
2. useLayouEffect
3. Browser rendering
4. useEffect

The formal document said. It's good to use for controlling `HTML` animation, scroll and etc... It's happend before real component's change. (Change which user can see) `useEffect` is happend after `useLayoutEffect`.

**I almost forget one thing. There is one more thing difference. `useLayoutEffect` works as a synchronous function.**

**Comment by chaminju927** [comment]
Is there any realtionship between synchronous function and HTML animation?

**Reply by honaseung** [reply]
There is no relationship between both. But the processing of rendering, `useLayoutEffect` is always worked before `HTML` rendering.
